synapse/storage/databases/state/store.py

```diff
@@ -404,6 +406,112 @@ def _insert_into_cache(
                 fetched_keys=non_member_types,
             )
 
+    async def store_state_deltas_for_batched(
+        self,
+        events_and_context: List[Tuple[EventBase, EventContext]],
+        room_id: str,
+        prev_group: int,
+    ) -> List[int]:
+        """Generate and store state deltas for a group of events and contexts created to be
+        batch persisted.
+
+        Args:
+            events_and_context: the events to generate and store a state groups for
+            and their associated contexts
```

Comment:
Can we mention that we only accept state events and that we modify the event contexts that are passed in?

```suggestion
        """Generate and store state deltas for a group of state events and contexts
        created to be batch persisted.

        Args:
            events_and_context: the events to generate and store state groups for and
                their associated contexts. Must only contain state events. The event
                contexts will be updated with the newly created state groups.
```